board/drivers/bootkick.h

```diff
@@ -0,0 +1,54 @@
+bool bootkick_ign_prev = false;
+BootState boot_state = BOOT_BOOTKICK;
+uint8_t bootkick_harness_status_prev = HARNESS_STATUS_NC;
+
+uint8_t boot_reset_countdown = 0;
+uint8_t waiting_to_boot_countdown = 0;
+bool bootkick_reset_triggered = false;
+
+void bootkick_tick(bool ignition, bool recent_heartbeat) {
+  BootState boot_state_prev = boot_state;
+  const bool harness_inserted = (harness.status != bootkick_harness_status_prev) && (harness.status != HARNESS_STATUS_NC);
+
+  if ((ignition && !bootkick_ign_prev) || harness_inserted) {
+    // bootkick on rising edge of ignition or harness insertion
+    boot_state = BOOT_BOOTKICK;
+  } else if (recent_heartbeat) {
+    // disable bootkick once openpilot is up
+    boot_state = BOOT_STANDBY;
+  } else {
+
+  }
```

Comment:
This is missing `usb_enumerated` from the original `dos` logic, does that matter? This code that does a similar check says to fall back to USB enumeration where it's supported (not sure of the differences): https://github.com/commaai/panda/blob/cd0f0bd034dde6814b4f5aa12da96a6f5631d141/board/main.c#L275C16-L275C22

Reply:
https://github.com/commaai/panda/pull/1690#discussion_r1387449917